src/node/internal/crypto_hash.ts

```diff
@@ -173,4 +183,76 @@ Hash.prototype.digest = function(this: Hash, outputEncoding?: string): Buffer |
   }
 }
 
-export {Hash};
+///////////////////////////
+
+interface Hmac extends Transform {
+  [kHandle]: cryptoImpl.HmacHandle;
+  [kState]: _kState;
+}
+
+export function createHmac(hmac: string, key: ArrayLike | KeyObject | CryptoKey,
+                           options?: TransformOptions): Hmac {
+  return new Hmac(hmac, key, options);
+}
+
+let Hmac = function(this: Hmac, hmac: string, key: ArrayLike | KeyObject | cryptoImpl.CryptoKey,
+                    options?: TransformOptions): Hmac {
+  if (!(this instanceof Hmac))
+    return new Hmac(hmac, key, options);
+  validateString(hmac, 'hmac');
+  const encoding = getStringOption(options, 'encoding');
+
+  if (key instanceof KeyObject) {
+    if (key.type !== 'secret')
+      throw new ERR_CRYPTO_INVALID_KEY_OBJECT_TYPE(key.type, 'secret');
+    this[kHandle] = new cryptoImpl.HmacHandle(hmac, key[kHandle]);
+  } else if (isCryptoKey(key)) {
+    if ((key as cryptoImpl.CryptoKey).type !== 'secret')
+      throw new ERR_CRYPTO_INVALID_KEY_OBJECT_TYPE((key as cryptoImpl.CryptoKey).type, 'secret');
+      this[kHandle] = new cryptoImpl.HmacHandle(hmac, key);
```

Comment:
As written above, please wrap the if branch in `{}`, because this indentation implies this statement is part of the if branch when it's not, better to be explicit.